### PR TITLE
Bug/ds names/check

### PR DESCRIPTION
**Pull request by KutluOzel-b**
## Proposed changes

https://github.com/zowe/vscode-extension-for-zowe/issues/1849

Changelog:

Added validation check while creating, renaming and using allocate alike feature for datasets

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [ x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x ] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x ] There is coverage for the code that I have added
- [ x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

References issues needed to be check if this also resolves them :

https://github.com/zowe/vscode-extension-for-zowe/issues/1769
https://github.com/zowe/vscode-extension-for-zowe/issues/1704
https://github.com/zowe/vscode-extension-for-zowe/issues/1511

### CODE REVIEW

**Comment by JillieBeanSim** [comment]
Hey @KutluOzel-b thanks for working on this.  Would you be able to write some unit tests to get the codecov/patch passing please?

**Comment by KutluOzel-b** [comment]
> Hey @KutluOzel-b thanks for working on this. Would you be able to write some unit tests to get the codecov/patch passing please?

Hi @JillieBeanSim  Thank you for the feedback 😋  . This problem caused by adding validator to inputBox. which can only be triggered by vscode api itself. I have tried to mock it ( what actually needed to be done is ; mock an api attribute function of an api function ) but couldn't find any proper solution for it. 

Besides that I just changed deprecated methods of inputBox due to low maintainability of unit tests of those functions. Please take this into consideration while doing tests.

**Review by JillieBeanSim — DISMISSED** [review]
Hey @KutluOzel-b this is a great validator and saw I had support for the special chars in the names.  I did see one issue with copy/past of ds member the validator isn't kicking in there

Screen Shot 2022-11-14 at 12 46 07 PM

and a lint issue
Screen Shot 2022-11-14 at 12 48 32 PM

File: packages/zowe-explorer/src/dataset/DatasetTree.ts [hunk 1]
```diff
@@ -1151,8 +1152,15 @@ export class DatasetTree extends ZoweTreeProvider implements IZoweTree<IZoweData
...
         const beforeMemberName = node.label as string;
         const dataSetName = node.getParent().getLabel() as string;
-        const options: vscode.InputBoxOptions = { value: beforeMemberName };
-        let afterMemberName = await UIViews.inputBox(options);
+        const options: vscode.InputBoxOptions = {
+            value: beforeMemberName,
+            validateInput: (text) => {
+                return validateDataSetName(text) === true
+                    ? null
+                    : localize("dataset.validation", "Enter valid dataset name");
```
Comment (t1m0thyj) [inline packages/zowe-explorer/src/dataset/DatasetTree.ts:1160]: Should this say "Enter valid member name" instead, and perform different validation? For example, `IBMUSER.TEST` is a valid data set name but not a valid member name.

File: packages/zowe-explorer/src/dataset/actions.ts [hunk 2]
```diff
@@ -392,8 +402,13 @@ export async function createMember(
 ) {
     const options: vscode.InputBoxOptions = {
         placeHolder: localize("createMember.inputBox.placeholder", "Name of Member"),
+        validateInput: (text) => {
+            return dsUtils.validateDataSetName(text) === true
+                ? null
+                : localize("dataset.validation", "Enter valid dataset name");
```
Comment (t1m0thyj) [inline packages/zowe-explorer/src/dataset/actions.ts:408]: Should this say "Enter valid member name" instead, and perform different validation than for data sets?

File: packages/zowe-explorer/src/dataset/actions.ts [hunk 3]
```diff
@@ -1266,8 +1288,13 @@ export async function pasteMember(
             const inputBoxOptions: vscode.InputBoxOptions = {
                 value: beforeMemberName,
                 placeHolder: localize("pasteMember.inputBox.placeHolder", "Name of Data Set Member"),
+                validateInput: (text) => {
+                    return dsUtils.validateDataSetName(text) === true
+                        ? null
+                        : localize("dataset.validation", "Enter valid dataset name");
```
Comment (t1m0thyj) [inline packages/zowe-explorer/src/dataset/actions.ts:1294]: Should this say "Enter valid member name" instead, and perform different validation than for data sets?

**Review by t1m0thyj — COMMENTED** [review]
Validation of data set names looks good to me 👍 

Left some questions regarding validation of member names - do we need a separate validator function for them that allows only 8 characters instead of 44?

File: packages/zowe-explorer/src/dataset/actions.ts (hunk 2 above)
Comment (JillieBeanSim) [reply packages/zowe-explorer/src/dataset/actions.ts]: addressed in commit [44418ad](https://github.com/zowe/vscode-extension-for-zowe/pull/1977/commits/44418add0d80197dbffdd8bd4f55121ae869e315)

File: packages/zowe-explorer/src/dataset/DatasetTree.ts (hunk 1 above)
Comment (JillieBeanSim) [reply packages/zowe-explorer/src/dataset/DatasetTree.ts]: addressed in commit [44418ad](https://github.com/zowe/vscode-extension-for-zowe/pull/1977/commits/44418add0d80197dbffdd8bd4f55121ae869e315)

**Comment by JillieBeanSim** [comment]
> Validation of data set names looks good to me 👍
> 
> Left some questions regarding validation of member names - do we need a separate validator function for them that allows only 8 characters instead of 44?

Good catch @t1m0thyj I have addressed the comments for separate test for members and updated strings for members in commit [44418ad](https://github.com/zowe/vscode-extension-for-zowe/pull/1977/commits/44418add0d80197dbffdd8bd4f55121ae869e315)

**Review by rudyflores — APPROVED** [review]
LGTM! seems to be working as intended :) Thanks @KutluOzel-b

**Review by t1m0thyj — APPROVED** [review]
LGTM! Thanks @KutluOzel-b and @JillieBeanSim 😀

**Pull request merged**